Jenkinsfile

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent any
+
+  parameters {
+    booleanParam(defaultValue: false, description: '', name: 'runEndToEndTestsOnPR')
+  }
+
+  options {
+    ansiColor('xterm')
+    timestamps()
+  }
+
+  libraries {
+    lib("pay-jenkins-library@master")
+  }
+
+  environment {
+    DOCKER_HOST = "unix:///var/run/docker.sock"
+    RUN_END_TO_END_ON_PR = "${params.runEndToEndTestsOnPR}"
+    JAVA_HOME="/usr/lib/jvm/java-1.11.0-openjdk-amd64"
+  }
+
+  stages {
+    stage('Maven Build') {
+      steps {
+        script {
+          long stepBuildTime = System.currentTimeMillis()
+          sh 'mvn -version'
+          sh 'mvn clean verify'
+          runProviderContractTests()
+          postSuccessfulMetrics("ledger.maven-build", stepBuildTime)
+        }
+      }
+      post {
+        failure {
+          postMetric("ledger.maven-build.failure", 1)
+        }
+      }
+    }
+
+    stage('Docker Build') {
+      steps {
+        script {
+          buildAppWithMetrics{
+            app = "ledger"
+          }
+        }
+      }
+      post {
+        failure {
+          postMetric("ledger.docker-build.failure", 1)
+        }
+      }
+    }
+    stage('Tests') {
```

Comment:
we don't need any e2e tests to start with as none of our current e2e tests will touch this service. We can change this as and when we write those tests

Reply:
i.e. you can remove this whole stage for now

Reply:
Thanks. No simple way to skip. Have removed for now. 👍